### PR TITLE
fix: Removed Internal Config from keploy config

### PR DESCRIPTION
**Pull request by ayush3160**
## Related Issue

Closes: #2076 

#### Describe the changes you've made
I have removed the internal config merging with default config because of which now keploy config only shows default config

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, local variables)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Please let us know if any test cases are added
NA

## Checklist:

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] New and existing unit tests pass locally with my changes.

## Screenshots (if any)
NA

### CODE REVIEW

**Comment by ayush3160** [comment]
I have read the CLA Document and I hereby sign the CLA

**Review by shivamsouravjha — CHANGES_REQUESTED** [review]
the build is failing, please look into it. Seems like the changes made are breaking the workflow.

**Comment by ayush3160** [comment]
@shivamsouravjha , I have checked locally and it is building successfully. Can you please help to find what could be the possible solution to this.

**Comment by shivamsouravjha** [comment]
* I tried this and had an issue. Our normal experience is  when we run keploy test a keploy config is created. But the issue is  when we run keploy test the config is that it has params~
```
generateGithubActions: true
keployContainer: keploy-v2
keployNetwork: keploy-network
cmdType: native
inCi: false
```

We don't want these to be present in the keploy.yml file as well. That part of solution doesn't seem to be implemented in the current PR.

**Comment by ayush3160** [comment]
@shivamsouravjha , I have done the changes for other commands also. Please have a look and let me know if any changes are needed.

**Review by shivamsouravjha — CHANGES_REQUESTED** [review]
I tried running your changes locally with docker container and here is what I found ~ 
image
image

Please test the case of docker image as well.

**Review by shivamsouravjha — APPROVED** [review]
LGTM!

File: cli/provider/cmd.go [hunk 1]
```diff
@@ -616,7 +617,11 @@ func (c *CmdConfigurator) ValidateFlags(ctx context.Context, cmd *cobra.Command)
 func (c *CmdConfigurator) CreateConfigFile(ctx context.Context, defaultCfg config.Config) error {
 	defaultCfg = c.UpdateConfigData(defaultCfg)
 	toolSvc := tools.NewTools(c.logger, nil)
-	configData := defaultCfg
+	configData, err := c.RemoveInternalParameters(defaultCfg)
```
Comment (charankamarapu) [inline cli/provider/cmd.go:620]: you can combine RemoveInternalParameters and yaml marshal into one function.

File: cli/provider/cmd.go [hunk 2]
```diff
@@ -644,3 +649,21 @@ func (c *CmdConfigurator) UpdateConfigData(defaultCfg config.Config) config.Conf
...
+func (c *CmdConfigurator) RemoveInternalParameters(defaultCfg config.Config) (map[string]interface{}, error) {
+	var cfgMap map[string]interface{}
+	err := mapstructure.Decode(defaultCfg, &cfgMap)
+
+	if err != nil {
+		utils.LogError(c.logger, err, "error decoding struct")
+		return nil, errors.New("error decoding struct")
+	}
+
+	fieldsToRemove := []string{"generateGithubActions", "keployNetwork", "keployContainer", "inCi", "cmdType"}
```
Comment (charankamarapu) [inline cli/provider/cmd.go:662]: For this you can create a function in config package itself. Once user adds something in internal config it should automatically be deleted from the config. We should not mention it again in another place

**Review by charankamarapu — CHANGES_REQUESTED** [review]
Please address the comments. Rest looks fine to me.

File: cli/provider/cmd.go (hunk 1 above)
Comment (ayush3160) [reply cli/provider/cmd.go]: Done

File: config/default.go [hunk 3]
```diff
@@ -64,6 +64,8 @@ inDocker: false
 cmdType: "native"
 `
 
+var InternalConfigFields = []string{"generateGithubActions", "keployNetwork", "keployContainer", "inCi", "cmdType"}
+
 var config = &Config{}
```
Comment (shivamsouravjha) [inline config/default.go:69]: why make it global when we have a function to fetch it as a string? You can make this a local variable.

Comment (ayush3160) [reply config/default.go]: Done.

**Comment by ayush3160** [comment]
@shivamsouravjha , have done the changes please have a look.

File: config/default.go [hunk 4]
```diff
@@ -110,3 +110,10 @@ func mergeStrings(srcStr, destStr string, infer bool, mergeOptions yaml.MergeOpt
 
 	return result.String()
 }
+
+func GetInternalConfigFields() []string {
+
+	InternalConfigFields := []string{"generateGithubActions", "keployNetwork", "keployContainer", "inCi", "cmdType"}
+
+	return InternalConfigFields
```
Comment (shivamsouravjha) [inline config/default.go:118]: lets keep it `internalConfigFields` as its the standard across repo

Comment (ayush3160) [reply config/default.go]: Done

**Review by shivamsouravjha — APPROVED** [review]
LGTM!

**Review by shivamsouravjha — COMMENTED** [review]
If we've a case where in config we add a param
```	
Random        string       `json:"random" yaml:"random" mapstructure:"-"`
```

here in this case if we run this the value random won't be written in the keploy.yaml file because we're dependent on mapstructure. This can act as a blocker.

Since this value still has its yaml tag it should be written in the yaml file which currently isn't.

**Comment by gouravkrosx** [comment]
Closing due to activity.

**Pull request closed without merging**